elements/pfe-clipboard/src/pfe-clipboard.js

```diff
@@ -0,0 +1,173 @@
+import PFElement from "../../pfelement/dist/pfelement.js";
+
+class PfeClipboard extends PFElement {
+  static get tag() {
+    return "pfe-clipboard";
+  }
+
+  static get meta() {
+    return {
+      title: "Clipboard",
+      description: "Copy current URL to clipboard."
+    };
+  }
+
+  get templateUrl() {
+    return "pfe-clipboard.html";
+  }
+
+  get styleUrl() {
+    return "pfe-clipboard.scss";
+  }
+
+  static get events() {
+    return {
+      copied: `${this.tag}:copied`
+    };
+  }
+
+  // Declare the type of this component
+  static get PfeType() {
+    return PFElement.PfeTypes.Content;
+  }
+
+  static get properties() {
+    return {
+      iconHidden: {
+        title: "Icon Hidden",
+        type: Boolean,
+        attr: "icon-hidden"
+      }
+    };
+  }
+
+  static get slots() {
+    return {
+      icon: {
+        title: "Icon",
+        slotName: "pfe-clipboard--icon",
+        slotClass: "pfe-clipboard__icon",
+        slotId: "icon"
+      },
+      default: {
+        title: "Text",
+        slotName: "pfe-clipboard--text",
+        slotClass: "pfe-clipboard__text",
+        slotId: "text"
+      },
+      textSuccess: {
+        title: "Text Success",
+        slotName: "pfe-clipboard--text--success",
+        slotClass: "pfe-clipboard__text--success",
+        slotId: "text--success"
+      }
+    };
+  }
+
+  constructor() {
+    super(PfeClipboard, { type: PfeClipboard.PfeType });
+
+    this._icon = this.shadowRoot.querySelector(`#icon`);
+    this._text = this.shadowRoot.querySelector(`#text`);
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    // If you need to initialize any attributes, do that here
+
+    this.icon = this.querySelector(`[slot="${this.tag}--icon"]`);
```

Comment:
There is also a getSlot method you can call against PFElement that does a similar query. It returns this: 

```js
[...this.children].filter(child => child.hasAttribute("slot") && child.getAttribute("slot") === name);
```

where name is the slot name.

Reply:
Oh very cool!  I like that!  These are actually not even used so I'm going to go ahead and remove them.  But that's good to know in the future.